cudax/include/cuda/experimental/__execution/transform_sender.cuh

```diff
@@ -0,0 +1,128 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDAX_EXECUTION_TRANSFORM_SENDER
+#define __CUDAX_EXECUTION_TRANSFORM_SENDER
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__type_traits/conditional.h>
+#include <cuda/std/__type_traits/is_valid_expansion.h>
+
+#include <cuda/experimental/__detail/utility.cuh>
+#include <cuda/experimental/__execution/domain.cuh>
+#include <cuda/experimental/__execution/env.cuh>
+
+#include <cuda/experimental/__execution/prologue.cuh>
+
+namespace cuda::experimental::execution
+{
+struct _CCCL_TYPE_VISIBILITY_DEFAULT transform_sender_t
+{
+  template <class _Domain, class _Sndr, class... _Env>
+  using __transform_domain_t =
+    _CUDA_VSTD::_If<_CUDA_VSTD::_IsValidExpansion<__transform_sender_result_t, _Domain, _Sndr, _Env...>::value,
+                    _Domain,
+                    default_domain>;
+
+  enum class __strategy
+  {
+    __passthru,
+    __transform,
+    __transform_recurse
+  };
+
+  struct __transform_strategy_t
+  {
+    bool __nothrow_;
+    __strategy __strategy_;
+  };
+
+  template <class _Self, class _Domain, class _Sndr, class... _Env>
+  _CCCL_TRIVIAL_API static constexpr auto __get_transform_strategy() noexcept -> __transform_strategy_t
+  {
+    using __dom_t _CCCL_NODEBUG_ALIAS    = __transform_domain_t<_Domain, _Sndr, _Env...>;
+    using __result_t _CCCL_NODEBUG_ALIAS = __transform_sender_result_t<__dom_t, _Sndr, _Env...>;
+
+    if constexpr (_CUDA_VSTD::_IsSame<__result_t&&, _Sndr&&>::value)
```

Comment:
Are the && here necessary?

Reply:
I want them to compare equal if one is a T and the other is a T&&

Reply:
So is it explicitly the `T` and `T&&` because commonly we use `remove_reference_t` or `remove_cvref_t` for that